### PR TITLE
Compile pages to .next/static directory

### PR DESCRIPTION
**Pull request by timneutkens**
The way Next.js has always worked was by resolving:

```
// url path
_next/{buildid}/page/index.js
```

To 

```
// server path
.next/bundles/pages/index.js
```

As you can see there's a difference between the URL path and the path on the filesystem. 

This created some downsides:

- The files have to be served by custom logic in the Next.js server, so these files couldn't be served directly by another web server, e.g. nginx to improve the performance of serving files without having it hit the origin Node.js server. Note that this downside isn't the case when using https://now.sh with [the Now CDN](https://zeit.co/cdn) 
- Next export has to copy the directory separately
- 2 routes were responsible for serving page bundles. (1 for .js and 1 for .js.map)
- on demand entries logic was colocated with these 2 routes.

What this PR aims to do is remove the 2 routes + extra next export logic and serve the pages directly from `.next/static`.

The new path to pages is:

```
// url path
_next/static/{buildid}/pages/index.js
```

To 

```
// server path
.next/static/{buildid}/pages/index.js
```

This also partially solves another issue for when users run `next build` on the same server where production is running, as it won't override the existing files.

### CODE REVIEW

**Comment by timneutkens** [comment]
I've made the `buildId` in development `development` but I wonder if we can find a clearer name for this.

File: server/hot-reloader.js [hunk 1]
```diff
@@ -73,6 +78,38 @@ export default class HotReloader {
       return
     }
 
+    const handlePageBundleRequest = async (req, res, parsedUrl) => {
```
Comment (arunoda) [inline server/hot-reloader.js]: What's this function is doing?
May be add an comment.

**Review by arunoda — APPROVED** [review]
Overall this looks great.
I also add an comment where I need more info.

File: server/hot-reloader.js (hunk 1 above)
Comment (timneutkens) [reply server/hot-reloader.js]: Added 👍

**Comment by rashidul0405** [comment]
I wish for this for a long long time, Thanks @timneutkens.

**Comment by codinronan** [comment]
@timneutkens another problem you just solved on accident is service workers: now I can modify my SW generator to work in development as well (normally wouldn't do this - but there are 1 or 2 scenarios where it makes sense). Since the buildId is known now, the code that generates the paths doesn't need to check if it is production :)

**Pull request merged**